lib/libc/minimal/include/bsearch.h

```diff
@@ -0,0 +1,22 @@
+/* bsearch.h */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_BSEARCH_H_
+#define ZEPHYR_LIB_LIBC_MINIMAL_INCLUDE_BSEARCH_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern void *bsearch(const void *base, const void *key,
+		     size_t count, size_t size,
+		     int (*cmp)(const void *key, const void *element));
```

Comment:
It would probably make more sense to follow the [parameter order and naming](https://www.gnu.org/software/libc/manual/html_node/Array-Search-Function.html) from the GNU C library:

```
void * bsearch (const void *key, const void *array, size_t count, size_t size, comparison_fn_t compare)
```